src/Microsoft.PowerShell.Commands.Utility/commands/utility/group-object.cs

```diff
@@ -113,10 +116,10 @@ private static string BuildName(List<ObjectCommandPropertyValue> propValues)
             StringBuilder sb = new StringBuilder();
             foreach (ObjectCommandPropertyValue propValue in propValues)
             {
-                if (propValue != null && propValue.PropertyValue != null)
+                var propValuePropertyValue = propValue?.PropertyValue;
+                if (propValuePropertyValue != null)
```

Comment:
Is the check really needed?

Reply:
yes, it is dereferenced in the else clause.